packages/web/components/bridge/amount-and-review-screen.tsx

```diff
@@ -222,7 +224,9 @@ export const AmountAndReviewScreen = observer(
       },
     });
 
-    if (!selectedAssetDenom) return;
+    if (!selectedAssetDenom) {
+      throw new Error("Selected asset denom is undefined.");
```

Comment:
We should be careful with this. Perhaps we just close the flow. This caused the page to crash when I went back/forward in the browser (and I believe the query params were updated)

Reply:
This will show the error boundary and let the user restart the flow. I don’t think we should close the flow, as that could be unexpected and jarring for the user

Reply:
I think I noticed that it showed the global fallback screen, at least in the preview. It was from going back or forward in the browser history (I don't remember which)

Reply:
Try going to amount screen then going back in browser. Do you think we could set state directly instead of relying on exception handling for happy path functionality ?

<img width="834" alt="Screenshot 2024-08-02 at 3 51 35 PM" src="https://github.com/user-attachments/assets/9998b528-39ab-4a97-8dc5-fbf530648c34">